lib/autoproj/ops/build.rb

```diff
@@ -84,7 +89,28 @@ def force_build_packages(selected_packages, all_enabled_packages)
             def build_packages(all_enabled_packages, options = Hash.new)
                 Autobuild.do_rebuild = false
                 Autobuild.do_forced_build = false
-                Autobuild.apply(all_enabled_packages, "autoproj-build", ['build'], options)
+                begin
+                    Autobuild.apply(all_enabled_packages, "autoproj-build", ['build'], options)
+                ensure
+                    build_report(all_enabled_packages) if @report_dir
+                end
+            end
+
+            TEMPLATES_DIR = File.join(File.dirname(__dir__), 'templates')
+            REPORT_BASENAME = "build_report.json"
+
+            # The path to the report file
+            #
+            # @return [String,nil] the path, or nil if the report should not
+            #    be generated
+            def report_path
+                File.join(@report_dir, REPORT_BASENAME) if @report_dir
+            end
+
+            def build_report(package_list)
+                erb = File.read(File.join(TEMPLATES_DIR, "build_report_helper.erb"))
+                FileUtils.mkdir_p @report_dir
+                IO.write(report_path, ::ERB.new(erb, nil, '-').result(binding))
```

Comment:
I think this should be:

```ruby

packages = []
package_list.each do |pkg_name|
  pkg = ws.find_autobuild_package(pkg_name)
  packages << {  name: pkg.name,
                 import_invoked: pkg.import_invoked?,
                 prepare_invoked: pkg.prepare_invoked?,
                 build_invoked: pkg.build_invoked?,
                 failed: pkg.failed?,
                 imported: pkg.imported?,
                 prepared: pkg.prepared?,
                 built: pkg.built?
  }
end

build_report = JSON.pretty_generate({ build_report: { timestamp: Time.now }, packages: [packages] })
```

And no templates...

Reply:
Agreed. Should just be `json = package_list.map do ... end` ;-)

Reply:
Even better :P